module/VuFind/src/VuFind/Db/Service/TagServiceInterface.php

```diff
@@ -63,27 +65,26 @@ public function matchText(string $text): array;
     /**
      * Get tags associated with the specified resource.
      *
-     * @param string $id          Record ID to look up
-     * @param string $source      Source of record to look up
-     * @param int    $limit       Max. number of tags to return (0 = no limit)
-     * @param ?int   $list        ID of list to load tags from (null for no
-     * restriction,  true for on ANY list, false for on NO list)
-     * @param ?int   $user        ID of user to load tags from (null for all users)
-     * @param string $sort        Sort type ('count' or 'tag')
-     * @param ?int   $userToCheck ID of user to check for ownership (this will
-     * not filter the result list, but rows owned by this user will have an is_me
-     * column set to 1)
+     * @param string                                $id        Record ID to look up
+     * @param string                                $source    Source of record to look up
+     * @param int                                   $limit     Max. number of tags to return (0 = no limit)
+     * @param UserListEntityInterface|int|bool|null $listOrId  List entity/ID to load tags from (null for no
+     * restriction, true for on ANY list, false for on NO lists)
```

Comment:
Yeah, this was painful, and you've convinced me to do something about it now while I'm at it. I've created three different methods (one currently unused, but there for completeness) to cover the functionality of the original method. I think this is a lot more clear! I also renamed all of the `getForResource` stuff to `get*TagsForRecord` since we're fetching based on record ID + source rather than database resource ID, so I think using the word "Resource" is just confusing. If time permits, I'll try to open another PR later to rename the other getForResource methods for similar clarity.

Since this is quite a big change, I'm re-requesting review before I merge this.